sklearn/decomposition/_nmf.py

```diff
@@ -803,23 +918,30 @@ def _fit_multiplicative_update(X, W, H, beta_loss='frobenius',
     previous_error = error_at_init
 
     H_sum, HHt, XHt = None, None, None
-    for n_iter in range(1, max_iter + 1):
+
+    batches = gen_batches(n_samples, batch_size)
+    batches = itertools.cycle(batches)
+
+    for n_iter, batch in zip(
+        range(iter_offset + 1, iter_offset + max_iter + 1),
+        batches
+    ):
         # update W
         # H_sum, HHt and XHt are saved and reused if not update_H
         delta_W, H_sum, HHt, XHt = _multiplicative_update_w(
-            X, W, H, beta_loss, l1_reg_W, l2_reg_W, gamma,
-            H_sum, HHt, XHt, update_H)
-        W *= delta_W
-
+            X[batch], W[batch], H, beta_loss, l1_reg_W, l2_reg_W,
+            gamma, H_sum, HHt, XHt, update_H)
+        W[batch] *= delta_W
```

Comment:
In the paper they test with doing 1 or several iterations for the W update. Maybe we could add a parameter `n_iter_W` (defaults to 1) to allow more fine grained control to the user.